File: 3658-gcd-of-odd-and-even-sums/3658-gcd-of-odd-and-even-sums.cpp
class Solution {
public:
    int gcdOfOddEvenSums(int n) {
        int sumOdd = n*n;
        int sumEven = n*(n+1);
        while(sumEven!=0){
            int r = sumOdd % sumEven;
            sumOdd = sumEven;
            sumEven = r;
        }
        return sumOdd;
    }
};

File: 3658-gcd-of-odd-and-even-sums/3658-gcd-of-odd-and-even-sums.py
class Solution:
    def gcdOfOddEvenSums(self, n: int) -> int:
        sumOdd = n * n
        sumEven = n * (n+1)

        while (sumEven != 0):
            r = sumOdd % sumEven
            sumOdd = sumEven
            sumEven = r
        
        return sumOdd